tools/RAiDER/models/gmao.py

```diff
@@ -0,0 +1,171 @@
+        session = pydap.cas.urs.setup_session('username', 'password', check_url=url)
+        ds = pydap.client.open_url(url, session=session)
+        
+        q = ds['qv'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        p = ds['pl'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        t = ds['t'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        h = ds['h'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        
+        # calculate the lat, lon and mean h for each layer in the regular grid
+        hs = np.zeros(72)
+        for ii in range(72):
+            hs[ii] = np.mean(h[ii,:,:])
+        
+        lats = np.arange((-90 + lat_min_ind * self._lat_res), (-90 + (lat_max_ind+1) * self._lat_res), self._lat_res)
+        lons = np.arange((-180 + lon_min_ind * self._lon_res), (-180 + (lon_max_ind+1) * self._lon_res), self._lon_res)
+        
+        # restructure the 3-D lat/lon/h in regular grid
+        _lons = np.broadcast_to(lons[np.newaxis, np.newaxis, :],t.shape)
+        _lats = np.broadcast_to(lats[np.newaxis, :, np.newaxis],t.shape)
+        _hs = np.broadcast_to(hs[:, np.newaxis, np.newaxis],t.shape)
+        
+        # Re-structure everything from (heights, lats, lons) to (lons, lats, heights)
+        p = np.transpose(p)
+        q = np.transpose(q)
+        t = np.transpose(t)
+        h = np.transpose(h)
+        _lats = np.transpose(_lats)
+        _lons = np.transpose(_lons)
+        _hs = np.transpose(_hs)
+        
+        
+        # check this
+        # data cube format should be lats,lons,heights
+        p = p.swapaxes(0,1)
+        q = q.swapaxes(0,1)
+        t = t.swapaxes(0,1)
+        h = h.swapaxes(0,1)
+        _lats = _lats.swapaxes(0,1)
+        _lons = _lons.swapaxes(0,1)
+        _hs = _hs.swapaxes(0,1)
+        
+        
+        # For some reason z is opposite the others
+        p = np.flip(p, axis = 2)
+        q = np.flip(q, axis = 2)
+        t = np.flip(t, axis = 2)
+        h = np.flip(h, axis = 2)
+        _lats = np.flip(_lats, axis = 2)
+        _lons = np.flip(_lons, axis = 2)
+        _hs = np.flip(_hs, axis = 2)
+        
+
+        # interpolate (p,q,t) along the vertical axis from irregular grid to regular grid
+        from RAiDER.interpolator import interp_along_axis
+        p_intrpl = interp_along_axis(h, _hs, p, axis = 2)
+        q_intrpl = interp_along_axis(h, _hs, q, axis = 2)
+        t_intrpl = interp_along_axis(h, _hs, t, axis = 2)
+        
+
+
+        
+        # assign the regular-grid (lat/lon/h) variables
+        
+        self._p = p_intrpl
+        self._q = q_intrpl
+        self._t = t_intrpl
+        self._lats = _lats
+        self._lons = _lons
+#        self._xs = lons
+#        self._ys = lats
+#        self._zs = np.flip(hs)
```

Comment:
if not used would remove it.

Reply:
@dbekaert all right. Removed them for now. Although I have been waiting for @jlmaurer 's response on why these must be 3-D cubes, which contradicts to what Jeremy told me, i.e. they should be 1-D vectors.